### PR TITLE
Fixes and tests a couple edge cases for RPN and infix

### LINKED ISSUE #580: Bmad to cheetah conversion failing
There are a number of conversion issues for converting LCLS Bmad files to cheetah lattices. The last time this was done successfully was using cheetah v0.7.0, which of course is over a year old. I'd like to work with @jank324 and @amylizzle to fix this and then set up nightly runs of converting the LCLS lattice to cheetah to track any breaking changes to cheetah as they come up. The first issue is with https://github.com/desy-ml/cheetah/blob/ca4b77ffbb40991c7921260de849f5ce755c19aa/cheetah/converters/utils/rpn.py#L112 which breaks if there are multiple elements with properties in a single equation (ie. `dbsy54a[l]-dbsy54b[l]`). This can be addressed (potentially more robustly) by replacing evaluate_expression with:

```python
import re
import math
from asteval import Interpreter

def evaluate_expression_ast(expr: str, data: dict) -> float:
    # Pattern: anything up to [attr], stopping at [
    pattern = r'([^\s+\-*/\\\[\]\(\)]+)\[([^\]]+)\]'
    matches = re.findall(pattern, expr)
    
    placeholder_map = {}
    eval_expr = expr

    # Create a math interpreter with standard math functions
    aeval = Interpreter()
    for func in dir(math):
        if not func.startswith("_"):
            aeval.symtable[func] = getattr(math, func)

    # Replace each match with a safe placeholder
    for i, (name, attr) in enumerate(matches):
        value = data[name][attr]
        placeholder = f"__var{i}__"
        placeholder_map[placeholder] = value
        # Escape special chars for literal replacement
        lookup_str = re.escape(f"{name}[{attr}]")
        eval_expr = re.sub(lookup_str, placeholder, eval_expr)

    # Inject placeholder values into interpreter
    for placeholder, value in placeholder_map.items():
        aeval.symtable[placeholder] = value

    # Evaluate the resulting math expression
    result = aeval(eval_expr)
    return result
```

an additional issue that I've run into is with the overlay definition regex which is not matching the current lcls lattice files. I can provide a copy of them if needed for testing

### PR DESCRIPTION
**Pull request by amylizzle**
## Description
Adds a couple extra test cases and fixes for RPN and infix parsing, in order to get LCLS lattice working.
Partially fixes #580, though a couple of edits to the BMAD files are needed to get fully working:
- add a `use, <something>` statement
- define `setsp = 0` in `common.bmad` 

Essentially there were two problems: 
1 - a variable was not defined which caused variables which used it to be treated as strings, which then caused cascading parsing errors as things tried to math on strings and became strings themselves
2 - cheetah expects a use statement somewhere in bmad files, and there doesn't appear to be one in LCLS2_cumaster.bmad

## Motivation and Context

- [x] I have raised an issue to propose this change (required for new features and bug fixes) 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (_required for a bug fix or a new feature_).
- [ ] I have updated the documentation accordingly.
  
  
- [x] I have reformatted the code and checked that formatting passes (**required**).
- [x] I have have fixed all issues found by `flake8` (**required**).
- [x] I have ensured that all `pytest` tests pass (**required**).
- [x] I have run `pytest` on a machine with a CUDA GPU and made sure all tests pass (**required**).
- [x] I have checked that the documentation builds (**required**).

Note: We are using a maximum length of 88 characters per line.

### CODE REVIEW

File: cheetah/converters/utils/infix.py [hunk 1]
```diff
@@ -164,6 +169,11 @@ def _tokenize_expression(expression: str, context: dict) -> list[str]:
                 if char != "(" and current_token in context:
                     tokens.append(context[current_token])
                 else:
+                    try:
+                        _ = float(current_token)
+                    except ValueError:
+                        if current_token not in context:
+                            print(f"Could not find {current_token} in context")
```
Comment (jank324) [inline cheetah/converters/utils/infix.py]: I'm not sure I understand why you are doing this ... ?

File: cheetah/converters/utils/rpn.py [hunk 2]
```diff
@@ -12,7 +13,7 @@ def evaluate_expression(expression: str, context: dict | None = None) -> Any:
 
     stack = []
     stripped = expression.strip()
-    for token in stripped.split():
+    for token in list(filter(None, re.split(r"(\+|\-|\*|/|\^)|\s", stripped))):
```
[the remark this answers is not in the document]
Comment (amylizzle) [reply cheetah/converters/utils/rpn.py]: depends if sci notation is supported in Elegant's RPN I guess? I haven't seen any. 

I would also want to add a bunch of tests if support for it is added though - maybe better for a separate PR in future if needed?
